src/mlpack/methods/linear_regression/linear_regression_fit_main.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file methods/linear_regression/linear_regression_fit_main.cpp
```

Comment:
The name `fit()` here matches the Python/scikit API, but maybe we should consider calling it `_train` internally to match the `Train()` function that mlpack provides?  Then, in the Python bindings, we could have a mapping from names like `_train` to `fit()`.  What do you think?